dpnp/dparray.pyx

```diff
@@ -124,6 +127,12 @@ cdef class dparray:
 
     """
 
+    with warnings.catch_warnings():
```

Comment:
I think we want this warning to be user visible, not [suppress it](https://docs.python.org/3/library/warnings.html#temporarily-suppressing-warnings).

Reply:
No, it will not suppress, it will create a context which copies the current filter and will restore upon exit from the context, as I got from [here](https://docs.python.org/3/library/warnings.html#available-context-managers).
The idea was to modify the filter with enabling deprecation warnings within the context, but without impact on the global filter.

Reply:
> I think we want this warning to be user visible, not [suppress it](https://docs.python.org/3/library/warnings.html#temporarily-suppressing-warnings).

I am in the wrong here. The `filterwarnings` turns the filter on, and use of context manager ensures its use does not alter users's setting of warning filtering.